asammdf/blocks/mdf_v4.py

```diff
@@ -259,55 +259,59 @@ def __init__(self, name=None, version="4.10", **kwargs):
             self.version = version
             self.name = Path("new.mf4")
 
-    def _check_finalised(self):
+    def _check_finalised(self) -> bool:
         flags = self.identification["unfinalized_standard_flags"]
+        
         if flags & 1:
             message = (
                 f"Unfinalised file {self.name}:"
-                "Update of cycle counters for CG/CA blocks required"
+                " Update of cycle counters for CG/CA blocks required"
             )
 
             logger.warning(message)
-        elif flags & 1 << 1:
+        if flags & 1 << 1:
             message = f"Unfinalised file {self.name}: Update of cycle counters for SR blocks required"
 
             logger.warning(message)
-        elif flags & 1 << 2:
+        if flags & 1 << 2:
             message = f"Unfinalised file {self.name}: Update of length for last DT block required"
 
             logger.warning(message)
-        elif flags & 1 << 3:
+        if flags & 1 << 3:
             message = f"Unfinalised file {self.name}: Update of length for last RD block required"
 
             logger.warning(message)
-        elif flags & 1 << 4:
+        if flags & 1 << 4:
             message = (
                 f"Unfinalised file {self.name}:"
-                "Update of last DL block in each chained list"
-                "of DL blocks required"
+                " Update of last DL block in each chained list"
+                " of DL blocks required"
             )
 
             logger.warning(message)
-        elif flags & 1 << 5:
+        if flags & 1 << 5:
             message = (
                 f"Unfinalised file {self.name}:"
-                "Update of cg_data_bytes and cg_inval_bytes "
-                "in VLSD CG block required"
+                " Update of cg_data_bytes and cg_inval_bytes"
+                " in VLSD CG block required"
             )
 
             logger.warning(message)
-        elif flags & 1 << 6:
+        if flags & 1 << 6:
             message = (
                 f"Unfinalised file {self.name}:"
-                "Update of offset values for VLSD channel required "
-                "in case a VLSD CG block is used"
+                " Update of offset values for VLSD channel required"
+                " in case a VLSD CG block is used"
             )
 
             logger.warning(message)
 
+        return flags == 0
+
     def _read(self, mapped=False):
```

Comment:
this will have a big performance impact. 

The best case approach would be to read the and at the end perform the finalization by modifying the loaded file structure (using the blocks defined in v4_blocks). This would allow to read the file once and still maintain good performance, but it need refactoring on your side and also in the asammdf to read all blocks (some blocks are skipped now) to create the mapping of address -> block to finalization steps.